cmd/kubeadm/app/cmd/join.go

```diff
@@ -128,6 +124,7 @@ var (
 type joinOptions struct {
 	cfgPath               string
 	token                 string
+	kubeconfigPath        string
```

Comment:
I don't think we should add `kubeconfigPath` here on this PR neither to the `joinOptions` or the `joinData` structs.

On this diff we see that we had:

```
		ctx := map[string]string{
			"KubeConfigPath": kubeadmconstants.GetAdminKubeConfigPath(),
			"etcdMessage":    etcdMessage,
		}
```

and now we are doing:

```
				ctx := map[string]string{
					"KubeConfigPath": data.KubeConfigPath(),
					"etcdMessage":    etcdMessage,
				}
```

But I think we should still call to `kubeadmconstants.GetAdminKubeConfigPath()` and remove `kubeconfigPath` from both `joinData` and `joinOptions` for this PR specifically, unless I'm missing something.

Reply:
+1 on that. We don't need it, it's something new and specific to the phase at hand.